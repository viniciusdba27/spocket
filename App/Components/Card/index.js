import React from 'react'
import { Text, View, Image as RNImage } from 'react-native'
import { PropTypes } from 'prop-types'
import { Image } from 'react-native-elements'
import { Helpers, Fonts, Metrics, Colors } from 'App/Theme'

import lookup from 'country-code-lookup'
import { Flag } from 'react-native-svg-flagkit'

import Style from './styles.js'

const Card = ({ data }) => (
  <View style={[Helpers.fill, Metrics.verticalMargin, Metrics.horizontalMargin, Style.card]}>
    <View style={Style.imgWrapper}>
      <Image
        resizeMode="cover"
        style={Style.img}
        source={{
          uri: data.image_cover_url,
        }}
        PlaceholderContent={
          <RNImage
            resizeMode="cover"
            style={Style.img}
            source={{
              uri: data.image_cover_thumb_url,
            }}
          />
        }
      />
    </View>
    <View
      style={[Metrics.verticalPadding, Metrics.horizontalPadding, Helpers.fill, Style.description]}
    >
      <View style={Metrics.bottomMargin}>
        <Text style={[Fonts.medium]} numberOfLines={1}>
          {data.title}
        </Text>
      </View>

      <View style={[Helpers.row, Metrics.bottomMargin]}>
        <Text style={[Fonts.medium, Style.by]} numberOfLines={1}>
          By
        </Text>
        <Text style={[Fonts.medium, Fonts.bold, Style.supplier]} numberOfLines={1}>
          {data.supplier_shop_name}
        </Text>
        <View style={[Style.flag, Metrics.smallHorizontalMargin]}>
          <Flag id={lookup.byCountry(data.country_origin).iso2} size={0.08} />
        </View>
      </View>
      <View style={[Helpers.fillCol]}>
        <View style={Helpers.fillRow}>
          <View style={[Helpers.fill, Style.price]}>
            <Text style={Fonts.small}>Price</Text>
            <Text style={[Fonts.regular, Fonts.bold]}>{data.formatted_price}</Text>
          </View>
          <View style={[Helpers.fill, Helpers.crossEnd]}>
            <Text style={[Fonts.small, { color: Colors.gray2 }]}>Retail Price</Text>
            <Text style={[Fonts.regular, Fonts.bold, { color: Colors.gray2 }]}>
              {data.formatted_msrp}
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
)

Card.propTypes = {
  data: PropTypes.any, // TODO: check for data shape
}

export default Card
