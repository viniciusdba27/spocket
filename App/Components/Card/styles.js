import { StyleSheet } from 'react-native'
import { Colors } from 'App/Theme'

export default StyleSheet.create({
  by: {
    alignSelf: 'flex-start',
    marginRight: 5,
  },
  card: {
    borderColor: Colors.gray1,
    borderRadius: 4,
    borderWidth: 1,
    height: 450,
  },
  description: {
    maxHeight: 130,
  },
  flag: {
    marginTop: 3,
  },
  img: {
    height: '100%',
    width: '100%',
  },
  imgWrapper: {
    flex: 2,
  },
  price: {
    marginTop: 3,
  },
  supplier: {
    alignSelf: 'flex-start',
    color: Colors.primary,
  },
})
