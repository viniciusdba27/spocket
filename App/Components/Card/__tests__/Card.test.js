import 'react-native'
import React from 'react'
import Card from '../index'
import renderer from 'react-test-renderer'

jest.useFakeTimers()
const data = {
  image_cover_thumb_url:
    'https://d2nxps5jx3f309.cloudfront.net/listing_images/attachments/fd2/c0c/62-/thumb/gl222.jpg',
  image_cover_url:
    'https://d2nxps5jx3f309.cloudfront.net/listing_images/attachments/fd2/c0c/62-/card/gl222.jpg',
  title: 'GL222 IP Rose Gold(Ion Plating) Brass Ring with',
  country_origin: 'United States',
  formatted_price: '$10.73 USD',
  formatted_msrp: '$26.82 USD',
}

it('renders correctly', async () => {
  const component = renderer.create(<Card data={data} />).toJSON()
  expect(component).toMatchSnapshot()
})
