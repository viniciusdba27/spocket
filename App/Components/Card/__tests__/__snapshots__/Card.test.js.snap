// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "marginVertical": 15,
      },
      Object {
        "marginHorizontal": 15,
      },
      Object {
        "borderColor": "#e1e1e1",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 450,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://d2nxps5jx3f309.cloudfront.net/listing_images/attachments/fd2/c0c/62-/card/gl222.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
        transition={true}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="RNE__Image__placeholder"
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://d2nxps5jx3f309.cloudfront.net/listing_images/attachments/fd2/c0c/62-/thumb/gl222.jpg",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "paddingVertical": 15,
        },
        Object {
          "paddingHorizontal": 15,
        },
        Object {
          "flex": 1,
        },
        Object {
          "maxHeight": 130,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        GL222 IP Rose Gold(Ion Plating) Brass Ring with
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginBottom": 15,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Object {
              "alignSelf": "flex-start",
              "marginRight": 5,
            },
          ]
        }
      >
        By
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "flex-start",
              "color": "#8144e5",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "marginTop": 3,
            },
            Object {
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={12}
            bbWidth={16.8}
            focusable={false}
            height={12}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 12,
                  "width": 16,
                },
              ]
            }
            vbHeight={15}
            vbWidth={21}
            width={16.8}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1,
                      1,
                      -986896,
                    ]
                  }
                  gradientTransform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  gradientUnits={0}
                  name="US_svg__a"
                  x1="50%"
                  x2="50%"
                  y1="0%"
                  y2="100%"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -3133628,
                      1,
                      -5167817,
                    ]
                  }
                  gradientTransform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  gradientUnits={0}
                  name="US_svg__b"
                  x1="50%"
                  x2="50%"
                  y1="0%"
                  y2="100%"
                />
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -12171649,
                      1,
                      -12829587,
                    ]
                  }
                  gradientTransform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  gradientUnits={0}
                  name="US_svg__c"
                  x1="50%"
                  x2="50%"
                  y1="0%"
                  y2="100%"
                />
              </RNSVGDefs>
              <RNSVGGroup
                fill={null}
                fillOpacity={1}
                fillRule={0}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGPath
                  d="M0 0h21v15H0z"
                  fill={
                    Array [
                      1,
                      "US_svg__a",
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M0 0h21v1H0V0zm0 2h21v1H0V2zm0 2h21v1H0V4zm0 2h21v1H0V6zm0 2h21v1H0V8zm0 2h21v1H0v-1zm0 2h21v1H0v-1zm0 2h21v1H0v-1z"
                  fill={
                    Array [
                      1,
                      "US_svg__b",
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M0 0h9v7H0z"
                  fill={
                    Array [
                      1,
                      "US_svg__c",
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
                <RNSVGPath
                  d="M1.5 2a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-5 1a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm1 1a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm1 1a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm1 1a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1zm-2 0a.5.5 0 1 1 0-1 .5.5 0 0 1 0 1z"
                  fill={
                    Array [
                      1,
                      "US_svg__a",
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginTop": 3,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 11,
              }
            }
          >
            Price
          </Text>
          <Text
            style={
              Array [
                undefined,
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            $10.73 USD
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "alignItems": "flex-end",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 11,
                },
                Object {
                  "color": "#7e768f",
                },
              ]
            }
          >
            Retail Price
          </Text>
          <Text
            style={
              Array [
                undefined,
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#7e768f",
                },
              ]
            }
          >
            $26.82 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
